dispatcher/hub/commonHub.go

```diff
@@ -99,18 +98,22 @@ func (wh *commonHub) handleBroadcast(events []data.Event) {
 }
 
 func (wh *commonHub) registerDispatcher(d dispatcher.EventDispatcher) {
+	wh.rwMut.Lock()
+	defer wh.rwMut.Unlock()
+
 	if _, ok := wh.dispatchers[d.GetID()]; ok {
 		return
 	}
-	wh.rwMut.Lock()
-	defer wh.rwMut.Unlock()
+
 	wh.dispatchers[d.GetID()] = d
 }
 
 func (wh *commonHub) unregisterDispatcher(d dispatcher.EventDispatcher) {
 	wh.rwMut.Lock()
-	defer wh.rwMut.Unlock()
 	if _, ok := wh.dispatchers[d.GetID()]; ok {
 		delete(wh.dispatchers, d.GetID())
 	}
+	wh.rwMut.Unlock()
```

Comment:
why did you not use it as a defer like on registerDispatched?

Reply:
fixed